packages/db-kit/src/utils/queryCompilation.ts

```diff
@@ -0,0 +1,113 @@
+import gql from "graphql-tag";
+import { Db, Resources, Process } from "@truffle/db";
+
+export interface QueryCompilationOptions {
+  db: Db;
+  project: Resources.IdObject<"projects">;
+  address: string;
+  network: Pick<Resources.Input<"networks">, "name">;
+}
+
+export async function queryCompilation({
+  db,
+  project,
+  address,
+  network
+}: QueryCompilationOptions): Promise<
+  Resources.Resource<"compilations"> | undefined
+> {
+  const ContractInstanceForAddress = gql`
+    query ContractInstanceForAddress(
+      $projectId: ID!
+      $network: ResourceNameInput!
+      $address: Address!
+    ) {
+      project(id: $projectId) {
+        contractInstance(address: $address, network: $network) {
+          contract {
+            compilation {
+              id
+
+              compiler {
+                name
+                version
+              }
+
+              sourceMaps {
+                bytecode {
+                  id
+                }
+                data
```

Comment:
For a minute I thought this was wrong. Which reminded me we should maybe update the data model at https://trufflesuite.github.io/artifact-updates/data-model.html. Or maybe we aren't using it anymore? I can't remember but want to put this here in case we should update it...

Reply:
Ahhh yeah. It'd be nice to update the data model, but maybe not before release? I barely use it anymore; the typeDefs have really become more canonical. What do you think?